api/cluster/controller.go

```diff
@@ -250,6 +252,33 @@ func (c *controller) Deploy(ctx context.Context, modelService *models.Service) (
 	}
 
 	inferenceURL := models.GetInferenceURL(s.Status.URL, isvcName, modelService.Protocol)
+
+	// Deploy virtual service
```

Comment:
Nit: `// Create / update virtual service` may be clearer since we are handling both here.